src/main/java/school/lemon/changerequest/java/io/pr1/task3/CountEngineImp.java

```diff
@@ -0,0 +1,163 @@
+package school.lemon.changerequest.java.io.pr1.task3;
+
+import java.io.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by User on 03.02.2017.
+ */
+public class CountEngineImp implements CountEngine {
+
+    @Override
+    public long countSpecificCharacter(String filename, char c) throws FileNotFoundException {
+
+        long result = 0;
+        File file = new File(filename);
+        BufferedInputStream bis;
+        FileInputStream fis;
+        if (!file.exists()) throw new FileNotFoundException();
+        try {
+            fis = new FileInputStream(file);
+            bis = new BufferedInputStream(fis);
+            while (bis.available() > 0) {
```

Comment:
Not sure it's a correct condition. File can be blocked by operation system e.g. So the correct way to read file char-by-char is smth like:
```
        int r;
        while ((r = reader.read()) != -1) {
            char ch = (char) r;
            System.out.println("Do something with " + ch);
        }
```